### PR TITLE
Add stack trace, layering/module information into kernels

### PR DESCRIPTION
**Pull request by voznesenskym**
Example in triton py
```
@reduction_heuristics(size_hints=[16384, 128], reduction_hint=ReductionHint.OUTER, filename=__file__)
@triton.jit
def kernel(in_ptr0, in_ptr1, in_ptr2, in_ptr3, in_ptr4, out_ptr0, out_ptr1, xnumel, rnumel, XBLOCK : tl.constexpr, RBLOCK : tl.constexpr):
    xoffset = tl.program_id(0) * XBLOCK
    xindex = xoffset + tl.reshape(tl.arange(0, XBLOCK), [XBLOCK, 1])
    xmask = xindex < xnumel
    rbase = tl.reshape(tl.arange(0, RBLOCK), [1, RBLOCK])
    x0 = xindex % 768
    x1 = (xindex // 768)
    _tmp9 = tl.zeros([XBLOCK, RBLOCK], tl.float32) + 0
    x3 = xindex
    _tmp10 = tl.zeros([XBLOCK, RBLOCK], tl.float32) + 0
    for roffset in range(0, rnumel, RBLOCK):
        rindex = roffset + rbase
        rmask = rindex < rnumel
        r2 = rindex

        # ORIGIN:
        # call_function aten.sum.dim_IntList
        # {'stack_trace': 'Module stack: {'self_model': 'BertForMaskedLM', 'self_model_bert': 'BertModel', 'self_model_bert_encoder': 'BertEncoder', 'self_model_bert_encoder_layer_5': 'BertLayer', 'self_model_bert_encoder_layer_5__attention': 'BertAttention', 'self_model_bert_encoder_layer_5__attention_output': 'BertSelfOutput', 'self_model_bert_encoder_layer_5__attention_output_LayerNorm': 'LayerNorm'}
        #   File "/data/home/voz/miniconda/envs/torchdynamo3.10/lib/python3.10/site-packages/transformers/models/bert/modeling_bert.py", line 384, in forward
        #     hidden_states = self.LayerNorm(hidden_states + input_tensor)
        #   File "/data/home/voz/miniconda/envs/torchdynamo3.10/lib/python3.10/site-packages/transformers/models/bert/modeling_bert.py", line 432, in forward
        #     attention_output = self.output(self_outputs[0], hidden_states)
        #   File "/data/home/voz/miniconda/envs/torchdynamo3.10/lib/python3.10/site-packages/transformers/models/bert/modeling_bert.py", line 493, in forward
        #     self_attention_outputs = self.attention(
        #   File "/data/home/voz/miniconda/envs/torchdynamo3.10/lib/python3.10/site-packages/transformers/models/bert/modeling_bert.py", line 606, in forward
        #     layer_outputs = layer_module(
        #   File "/data/home/voz/miniconda/envs/torchdynamo3.10/lib/python3.10/site-packages/transformers/models/bert/modeling_bert.py", line 1017, in forward
        #     encoder_outputs = self.encoder(
        #   File "/data/home/voz/miniconda/envs/torchdynamo3.10/lib/python3.10/site-packages/transformers/models/bert/modeling_bert.py", line 1350, in forward
        #     outputs = self.bert(
        #   File "/scratch/voz/work/benchmark/torchbenchmark/util/framework/huggingface/model_factory.py", line 114, in train
        #     outputs = self.model(**self.example_inputs)
        # 
        # Gradient addition node due to mulitple use of tensor around:
        # Module stack: {'self_model': 'BertForMaskedLM', 'self_model_bert': 'BertModel', 'self_model_bert_embeddings': 'BertEmbeddings', 'self_model_bert_embeddings_word_embeddings': 'Embedding'}
        #   File "/data/home/voz/miniconda/envs/torchdynamo3.10/lib/python3.10/site-packages/transformers/models/bert/modeling_bert.py", line 235, in forward
        #     inputs_embeds = self.word_embeddings(input_ids)
        #   File "/data/home/voz/miniconda/envs/torchdynamo3.10/lib/python3.10/site-packages/transformers/models/bert/modeling_bert.py", line 1010, in forward
        #     embedding_output = self.embeddings(
        #   File "/data/home/voz/miniconda/envs/torchdynamo3.10/lib/python3.10/site-packages/transformers/models/bert/modeling_bert.py", line 1350, in forward
        #     outputs = self.bert(
        #   File "/scratch/voz/work/benchmark/torchbenchmark/util/framework/huggingface/model_factory.py", line 114, in train
        #     outputs = self.model(**self.example_inputs)
        # ', 'tensor_meta': TensorMetadata(shape=torch.Size([768]), dtype=torch.float32, requires_grad=False, stride=(1,), memory_format=torch.contiguous_format, is_quantized=False, qparams={})}
        # END ORIGIN

        tmp0 = tl.load(in_ptr0 + x0 + (768*r2) + (98304*x1), xmask & rmask, eviction_policy='evict_last')
        tmp1 = tl.load(in_ptr1 + x0 + (768*r2) + (98304*x1), xmask & rmask, eviction_policy='evict_last')
        tmp3 = tl.load(in_ptr2 + x0 + (768*r2) + (98304*x1), xmask & rmask, eviction_policy='evict_last')
        tmp4 = tl.load(in_ptr3 + r2 + (128*x1), xmask & rmask, eviction_policy='evict_last')
        tmp6 = tl.load(in_ptr4 + r2 + (128*x1), xmask & rmask, eviction_policy='evict_last')
        tmp2 = tmp0 + tmp1
        tmp5 = tmp3 - tmp4
        tmp7 = tmp5 * tmp6
        tmp8 = tmp2 * tmp7
        _tmp9 = tl.where(xmask & rmask, _tmp9 + tmp8, _tmp9)
        _tmp10 = tl.where(xmask & rmask, _tmp10 + tmp2, _tmp10)
    tmp9 = tl.reshape(tl.sum(_tmp9, 1), [XBLOCK, 1])
    tl.store(out_ptr0 + x3, tmp9, xmask)
    tmp10 = tl.reshape(tl.sum(_tmp10, 1), [XBLOCK, 1])
    tl.store(out_ptr1 + x3, tmp10, xmask)

```

### CODE REVIEW

File: torchinductor/scheduler.py [hunk 1]
```diff
@@ -183,6 +184,41 @@ def can_free(self):
...
+        for o in origins:
+            if o.op == "output":
+                # These are boring and samey
+                continue
+
+            out_lines.append("")
+            # TODO(voz): Should the pragma be constant somewhere?
+            out_lines.append("#pragma CMT ORIGIN:")
+            out_lines.append(f"#pragma CMT {o.op} {o.target}")
+            out_line_meta = f"{o.meta}"
```
Comment (SherlockNoMad) [inline torchinductor/scheduler.py:205]: dumping the entire meta might  be too verbose... 
you can look at append_stacktrace_summary's impl in graph.py if you just want to show the summary...

**Review by SherlockNoMad — APPROVED** [review]

File: torchinductor/scheduler.py (hunk 1 above)
Comment (voznesenskym) [reply torchinductor/scheduler.py:205]: I like the whole meta, I think it's all useful. The whole thing is config disable-able anyway.

File: torchinductor/config.py [hunk 2]
```diff
@@ -49,6 +49,8 @@
 # replace small reductions with pointwise, disable with `= 1`
 unroll_reductions_threshold = 8
 
+comment_origin = True
```
Comment (jansel) [inline torchinductor/config.py:52]: We should turn this off by default because it could increase compile times by preventing cache hits at the triton level.

**Review by jansel — CHANGES_REQUESTED** [review]
The output looks too verbose to me.  Perhaps we could only show one level of stack trace (skipping torch files) and put it on a single line.

**Comment by voznesenskym** [comment]
Updated:

```
import triton
import triton.language as tl
from torchinductor.triton_ops.autotune import pointwise_heuristics

@pointwise_heuristics(size_hints=[2048, 1024], filename=__file__)
@triton.jit
def kernel(in_ptr0, out_ptr0, xnumel, ynumel, XBLOCK : tl.constexpr, YBLOCK : tl.constexpr):
    xoffset = tl.program_id(0) * XBLOCK
    xindex = xoffset + tl.reshape(tl.arange(0, XBLOCK), [XBLOCK, 1])
    xmask = xindex < xnumel
    yoffset = tl.program_id(1) * YBLOCK
    yindex = yoffset + tl.reshape(tl.arange(0, YBLOCK), [1, YBLOCK])
    ymask = yindex < ynumel
    x0 = xindex % 512
    x1 = (xindex // 512)
    y2 = yindex
    x3 = xindex

    # ORIGIN:
    # call_function aten.mm.default
    #    File "/scratch/voz/work/benchmark/torchbenchmark/util/framework/huggingface/model_factory.py", line 114, in train\    outputs = self.model(**self.example_inputs)    # END ORIGIN

    tmp0 = tl.load(in_ptr0 + x0 + (512*y2) + (393216*x1), xmask & ymask)
    tl.store(out_ptr0 + y2 + (768*x3) + tl.zeros([XBLOCK, YBLOCK], tl.int32), tmp0, xmask & ymask)
```

**Review by jansel — APPROVED** [review]
